### ISSUE TITLE
Single quoutes in gstreamer wrapper prevent evaluation of profile plugins path search

### ISSUE THREAD

**Issue by roosemberth** [NixOS/nixpkgs#70525]
Description: what the title says.

> `cat "$(nix-store -r "$(nix-instantiate '<nixpkgs>' -A gst_all_1.gstreamer.dev)")"/bin/gst-launch-1.0`

```
#! /nix/store/...-bash-4.4-p23/bin/bash -e
export GST_PLUGIN_SYSTEM_PATH=$GST_PLUGIN_SYSTEM_PATH${GST_PLUGIN_SYSTEM_PATH:+':'}'$(unset _tmp; for profile in $NIX_PROFILES; do _tmp=$profile/lib/gstreamer-1.0${_tmp:+:}$_tmp; done; printf $_tmp)'
exec -a "$0" "/nix/store/...-gstreamer-1.14.4-dev/bin/.gst-launch-1.0-wrapped"  "${extraFlagsArray[@]}" "$@"
```

There are single quotes around the plugin search in NIX_PROFILES, preventing it from evaluation on runtime. (i.e. `'$(unset _tmp; ... printf $_tmp)'`, instead of `"$(unset _tmp; ... printf $_tmp)"`)

**Steps to reproduce**

> `nix-env -iA nixos.gst_all_1.gstreamer`
> `nix-env -iA nixos.gst_all_1.gst-plugins-good`
> `gst-launch-1.0 pulsesrc ! fakesink`

```
ERROR: pipeline could not be constructed: no element "pulsesrc".
```

(Same behavior can be achieved by placing `gst_all_1.gstreamer` and `gst_all_1.gst-plugins-good` in your `configuration.nix`)

**Expected behavior**

> `cat "$(nix-store -r "$(nix-instantiate '<nixpkgs>' -A gst_all_1.gstreamer.dev)")"/bin/gst-launch-1.0`

```
#! /nix/store/...-bash-4.4-p23/bin/bash -e
export GST_PLUGIN_SYSTEM_PATH=$GST_PLUGIN_SYSTEM_PATH${GST_PLUGIN_SYSTEM_PATH:+':'}"$(unset _tmp; for profile in $NIX_PROFILES; do _tmp=$profile/lib/gstreamer-1.0${_tmp:+:}$_tmp; done; printf $_tmp)"
exec -a "$0" "/nix/store/...-gstreamer-1.14.4-dev/bin/.gst-launch-1.0-wrapped"  "${extraFlagsArray[@]}" "$@"
```

> `nix-env -iA nixos.gst_all_1.gstreamer`
> `nix-env -iA nixos.gst_all_1.gst-plugins-good`
> `gst-launch-1.0 pulsesrc ! fakesink`

```
Setting pipeline to PAUSED ...
Pipeline is live and does not need PREROLL ...
Setting pipeline to PLAYING ...
New clock: GstPulseSrcClock
Redistribute latency...
...
```

**Additional context**

Evaluation trace:

`gst-launch-1.0`: https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/gstreamer/core/default.nix#L63
`wrapProgram`: https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/setup-hooks/make-wrapper.sh#L139
`makeWrapper`: https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/setup-hooks/make-wrapper.sh#L33

Workaround: Steal another derivation's (constructed with wrapGappsHook) environment via mkShell.inputsFrom. e.g.

> `nix-shell =(echo "with (import <nixpkgs> {}).pkgs; mkShell { inputsFrom = [mopidy]; }") --run "gst-launch-1.0 pulsesrc ! fakesink"`

Maintainer information:
```yaml
attribute: gst_all_1.gstreamer.dev
```

**Comment by hedning** [comment]
The quote issue was fixed in 9b7742eefc70b94c85c5942f2e4dc52a5b5f5cef

However `gst_all_1` sets `GST_PLUGIN_SYSTEM_PATH` instead of `GST_PLUGIN_SYSTEM_PATH_1_0` which makes the setup ineffective. Using the correct variable I'm able to find the pulseaudio plugin.

**Comment by roosemberth** [comment]
Thanks, closing the issue.
/close
